### ISSUE TITLE
How to access a n dim tensor

### ISSUE THREAD

**Issue by Flocksserver** [sonos/tract#942]
Hi there,
I'm a bit confused about the tract api. My model output has 2 logits. The first one has one dimension. Therefore, I can get the predicted value in the same way as shown in the examples. The second one has more dimensions. The to_array_view only allows types implement `Datum` - but there is no vec or array or anything like that.

```rust
    let outputs = model.run(tvec!(input_ids.into(), attention_mask.into())).unwrap();
    let intent_logits = outputs[0].to_array_view::<f32>().unwrap();
    let slot_logits = outputs[1].to_array_view::<f32>().unwrap(); // ??
```
If I print out slot_logits the following debug info is displayed:

```
[[[0.046216786, -0.04436322, 2.196459, -5.619391e-5, -0.5223904, ..., 0.18682338, 0.42898875, -0.75157386, -0.41718343, -0.9214676],
  [-0.43851694, -0.5855652, 8.87492, -0.6223176, -0.20071359, ..., -0.27404433, -0.032524835, -0.3375598, 0.047054436, -0.4387788],
  [-0.4443598, -0.6086989, 8.780408, -0.68458056, -0.30464306, ..., -0.2814015, -0.06829582, -0.3323328, 0.09056011, -0.40781564],
  [-0.41140473, -0.67001593, 8.832634, -0.6492824, -0.21625353, ..., -0.21029603, -0.1063809, -0.20338516, 0.01558703, -0.57910687],
  [-0.38607445, -0.65450966, 8.844382, -0.6497509, -0.19975904, ..., -0.2357347, 0.01696811, -0.24379173, 0.14299859, -0.41308415],
  ...,
  [-0.808787, -0.41400576, 5.26754, -0.43476292, -0.3451499, ..., 0.5746111, -0.02954241, -0.4906926, -0.18069722, -0.71241635],
  [-0.2621671, -0.3874972, 2.4530666, -0.4070676, -0.15413575, ..., -0.14476787, -0.13609557, -0.4087047, -0.33043995, -0.7413258],
  [-0.30875868, 0.1593123, 1.7369856, -0.38965702, -0.4194109, ..., 0.44680768, 0.19038537, -0.6701888, -0.4746719, -0.04854029],
  [-0.18919832, -0.041673094, 0.60368574, 0.3113171, -0.18765324, ..., -0.3121799, -0.2118019, -0.2171652, -0.24881445, -0.044289935],
  [-0.05159561, -0.049215008, -0.47322214, 0.3796391, -0.001572906, ..., -0.42708483, -0.2189663, 0.035680477, -0.344612, 0.09602491]]], shape=[1, 16, 122], strides=[1952, 122, 1], layout=Cc (0x5), dynamic ndim=3

```
Did I miss something? 

I know the length of each array inside the output. Therefore, an ugly ugly workaround would be to treat the output as a single array and slice it. But this is not really what I want 🙃  

Thanks 🙏

**Comment by Flocksserver** [comment]
That was a really really fast reply - thank you!
The dimensions and the values are as expected.

Now I use as_slice, get the one big `&[f32]` and chunk the 1952 values (16 x 122) into - obviously 16 chunks with 122 values. I thought this was too complicated and i missed something. I expected to be able to access the values with something like this: `my_tensor[0][0] // for the first 122 values out of 16`

Long story short - using as_slice with chunking works an I am happy with it 👍 

Kudos to you, (sonos - if I understood correctly, they allow open source work on this) and this project for bringing onnx inference to rust. 🙏

**Comment by kali** [comment]
Just for completion here...

The ndarray binding gives you access with indexing. it should looks like `output.to_array_view::<f32>()?[&[0,1,2]]`... You can also use ndarray .into_dimensionality() to "cast" the view to ArrayView3 instead of the ArrayViewD, which "knows" the rank (ndim) of the array, so you can access it with tuples instead of slices: `output.to_array_view::<f32>()?.into_dimensionality()?[(0,1,2)]`. (I have not tested these code bits, I know it doew work, the syntax may be slightly wrong).
